lib/ae_canary/fork_monitor/model/chain_walker.ex

```diff
@@ -0,0 +1,195 @@
+defmodule AeCanary.ForkMonitor.Model.ChainWalker do
+  require Logger
+
+  def nodes() do
+    ["http://206.81.24.215:3013/v2"]
+  end
+
+  @doc """
+   Configurable http client to allow simple test mocking - allow configuration
+   of the module used for the get! request used to fetch the chain.
+  """
+  def http_client() do
+    {:ok, v} = Application.fetch_env(:ae_canary, AeCanary.ForkMonitor)
+    Keyword.get(v, :fork_monitor_http_client)
+  end
+
+  def updateChainEnds(max_depth, notify_pid \\ nil) do
+    uniqueChainEnds = getChainEnds()
+    Logger.info("Found chain ends #{Enum.map(uniqueChainEnds, fn e -> e.hash end)}")
+    topHeight = Enum.map(uniqueChainEnds, fn e -> e.block["height"] end) |> Enum.max()
+    stopAtHeight = max(0, topHeight - max_depth)
+
+    Logger.debug("Found top height of #{topHeight}, stopping at height #{stopAtHeight}")
+
+    uniqueChainEnds = Enum.reject(uniqueChainEnds, fn e -> e.block["height"] < stopAtHeight end)
+
+    Enum.map(uniqueChainEnds, fn e ->
+      Logger.debug("Starting with chain end #{e.hash} (#{e.block["height"]})")
+    end)
+
+    notify(
+      notify_pid,
+      {:started_sync, Enum.map(uniqueChainEnds, fn e -> {e.hash, e.block["height"]} end),
+       topHeight}
+    )
+
+    danglingBranches =
+      AeCanary.ForkMonitor.Model.unattachedBlocks()
+      |> Enum.map(fn branch -> %{hash: branch.keyHash, nodeUrl: hd(nodes())} end)
```

Comment:
An idea: maybe scan a few nodes and aggregate results. This of course can be done in a separate PR.